#!/bin/bash
#PBS -N Filter_variants
#PBS -j oe
#PBS -m ae
#PBS -l nodes=1
#PBS -l ncpus=11
#PBS -l mem=50gb
#PBS -l walltime=24:00:00
#PBS -M jacob.westaway@menzies.edu.au

echo "---------------------------------------"
echo "PBS: Job identifier is $PBS_JOBID"
echo "PBS: Job name is $PBS_JOBNAME"

echo "---------------------------------------"
echo "---------------------------------------"
echo 'Change to working directory and set env variables'
OUTDIR="/home/jwestaway/pk_pipeline/ZB_100/outputs/vcf_consensus/filtering_tests/GATK_BP"
PICARD="/usr/local/miniconda3/pkgs/picard-2.21.9-0/share/picard-2.21.9-0/picard.jar"
INDEXTDIR="/home/jwestaway/pk_pipeline/ref_genomes/PKA1H1/fasta/strain_A1_H.1.Icor.fasta"
mkdir $OUTDIR
cd /home/jwestaway/pk_pipeline/ZB_100/outputs/vcf_consensus


echo "---------------------------------------"
echo 'filter snps'
/usr/local/jdk1.8.0_131/bin/java -Djava.iodir=$PBS_JOBFS -Xms3200m -Xmx3600m -jar /home/jwestaway/pk_pipeline/tools/gatk-4.2.2.0/gatk-package-4.2.2.0-local.jar \
    VariantFiltration \
    -V PK_consensus_corrected_snps.vcf \
    -filter "QD < 2.0" --filter-name "QD2" \
    -filter "QUAL < 30.0" --filter-name "QUAL30" \
    -filter "SOR > 3.0" --filter-name "SOR3" \
    -filter "FS > 60.0" --filter-name "FS60" \
    -filter "MQ < 40.0" --filter-name "MQ40" \
    -filter "MQRankSum < -12.5" --filter-name "MQRankSum-12.5" \
    -filter "ReadPosRankSum < -8.0" --filter-name "ReadPosRankSum-8" \
    -O $OUTDIR/snps_filtered.vcf.gz


echo "---------------------------------------"
echo 'filter indels'
/usr/local/jdk1.8.0_131/bin/java -Djava.iodir=$PBS_JOBFS -Xms3200m -Xmx3600m -jar /home/jwestaway/pk_pipeline/tools/gatk-4.2.2.0/gatk-package-4.2.2.0-local.jar \
    VariantFiltration \
    -V PK_consensus_corrected_indels.vcf \
    -filter "QD < 2.0" --filter-name "QD2" \
    -filter "QUAL < 30.0" --filter-name "QUAL30" \
    -filter "FS > 200.0" --filter-name "FS200" \
    -filter "ReadPosRankSum < -20.0" --filter-name "ReadPosRankSum-20" \
    -O $OUTDIR/indels_filtered.vcf.gz


echo "---------------------------------------"
echo 'Make diagnostic tables for Variants Scores - snps'

/usr/local/jdk1.8.0_131/bin/java -Djava.iodir=$PBS_JOBFS -Xms3200m -Xmx3600m -jar /home/jwestaway/pk_pipeline/tools/gatk-4.2.2.0/gatk-package-4.2.2.0-local.jar \
    VariantsToTable \
    -R $INDEXTDIR \
    -V $OUTDIR/snps_filtered.vcf.gz \
    -F CHROM -F POS -F QUAL -F QD -F DP -F MQ -F GQ -F MQRankSum -F FS -F ReadPosRankSum -F SOR \
    -O $OUTDIR/GVCFall_SNPs.table


echo "---------------------------------------"
echo 'Make diagnostic tables for Variants Scores - indels'

/usr/local/jdk1.8.0_131/bin/java -Djava.iodir=$PBS_JOBFS -Xms3200m -Xmx3600m -jar /home/jwestaway/pk_pipeline/tools/gatk-4.2.2.0/gatk-package-4.2.2.0-local.jar \
    VariantsToTable \
    -R $INDEXTDIR \
    -V $OUTDIR/indels_filtered.vcf.gz \
    -F CHROM -F POS -F QUAL -F QD -F DP -F MQ -F GQ -F MQRankSum -F FS -F ReadPosRankSum -F SOR \
    -O $OUTDIR/GVCFall_INDELs.table




echo "---------------------------------------"
echo 'Get summary stats for different tools and combined'
echo "---------------------------------------"

echo "---------------------------------------"
echo "SNPs"

echo "---------------------------------------"
echo "Exectute bcftools to select variants that pass a specific threshold: FMT/GQ>30 & FMT/DP>10"
bcftools query -i 'QUAL>30 & FMT/GQ>30 & FMT/DP>10' -f '%CHROM %POS %ID %REF %ALT [\t%SAMPLE DP=%DP GQ=%GQ MQ=%MQ PL=%PL]\n' merged_GATK_samtools_variants.vcf.gz > merged.tsv

echo "---------------------------------------"
echo "Execute R script to get consensus variants and stats"
source /usr/local/miniconda3/etc/profile.d/conda.sh
conda activate R_3.6.2
Rscript /home/jwestaway/pk_pipeline/ZB_100/scripts/14_Consensus_vcf/vcf_wrangle.R
conda deactivate

echo "---------------------------------------"
echo "Indels"

echo "---------------------------------------"
echo "Exectute bcftools to select variants that pass a specific threshold: FMT/GQ>30 & FMT/DP>10"
bcftools query -i 'QUAL>30 & FMT/GQ>30 & FMT/DP>10' -f '%CHROM %POS %ID %REF %ALT [\t%SAMPLE DP=%DP GQ=%GQ MQ=%MQ PL=%PL]\n' merged_GATK_samtools_variants.vcf.gz > merged.tsv

echo "---------------------------------------"
echo "Execute R script to get consensus variants and stats"
source /usr/local/miniconda3/etc/profile.d/conda.sh
conda activate R_3.6.2
Rscript /home/jwestaway/pk_pipeline/ZB_100/scripts/14_Consensus_vcf/vcf_wrangle.R
conda deactivate

echo "---------------------------------------"
echo 'Finished'